src/sentry/event_manager.py

```diff
@@ -98,10 +101,12 @@ def get_grouping_behavior(event):
     return ('fingerprint', get_hashes_from_fingerprint_with_reason(event, fingerprint))
 
 
-def get_hashes_from_fingerprint(event, fingerprint):
-    default_values = set(['{{ default }}', '{{default}}'])
-    if any(d in fingerprint for d in default_values):
-        default_hashes = get_hashes_for_event(event)
+def _get_hashes_from_fingerprint(get_hash_inputs, fingerprint):
+    # get_hash_inputs should be a function with the event / data
+    # argument bound, for example
+    #   - functools.partial(get_hashes_for_event, event)
+    if any(d in fingerprint for d in DEFAULT_FINGERPRINT_VALUES):
+        default_hashes = get_hash_inputs()
```

Comment:
What was the reason that this seemed better to pass as a callable instead of a list?

Reply:
for the optimization of not generating the hashes if they aren't needed. i think i remember your suggesting that we only generate them when needed in a previous iteration of this PR

Reply:
Ah ok, this makes sense.